reviews/views.py

```diff
@@ -158,6 +185,37 @@ class ReviewableLogList(JSONAPIBaseView, generics.ListAPIView, DjangoFilterMixin
     def get_default_django_query(self):
         return Q(reviewable_id=self.get_preprint().id)
 
-    # overrides ListAPIView
+    # overrides ListCreateAPIView
+    def perform_create(self, serializer):
+        serializer.save(user=self.request.user, reviewable=self.get_preprint())
+
+    # overrides ListCreateAPIView
     def get_queryset(self):
-        return ReviewLog.objects.select_related('reviewable', 'reviewable__provider').filter(self.get_query_from_request()).distinct()
+        return review_logs_queryset().filter(self.get_query_from_request())
+
+
+class SetUpProvider(JSONAPIBaseView, generics.UpdateAPIView):
```

Comment:
I chatted with Abram, and we're going to have a PATCH to the preprint provider with a validator that says that you have to modify all those fields at once but they also have to be null. Documentation for developer.osf.io will describe this validation. Failing will result in a 409 that will include a good description of the problem and how to fix it.